sktime/transformations/base.py

```diff
@@ -1083,7 +1085,9 @@ def _scitype_A_higher_B(scitypeA, scitypeB):
                 msg, var_name=msg_X, allowed_msg=allowed_msg, raise_exception=True
             )
 
-        if DtypeKind.CATEGORICAL in X_metadata["feature_kind"]:
+        if DtypeKind.CATEGORICAL in X_metadata["feature_kind"] and not self.get_tag(
+            "capability:categorical_in_X"
+        ):
```

Comment:
error message is no longer accurate, this should point to the tag.

Also should point to the name of the transformer now, because it is specific to it.